ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/StatementExecutor.java

```diff
@@ -258,6 +267,10 @@ private String handleCreateAsSelect(
       final String statementStr,
       final boolean wasDropped
   ) {
+    if (QueryCapacityUtil.exceedsPersistentQueryCapacity(
```

Comment:
I think this is checking the capacity against the _current_ limit, not the one in the command.

Looks like you can move the check into the `startQuery` call, which is already building the adjusted `KsqlConfig`.